backend/matchmaking/views.py

```diff
@@ -23,19 +26,64 @@ def survey_results_create(request):
     """
     if request.method == 'POST':
         try:
+            # Grab data from request
             data = json.loads(request.body)
+            # Grab films from data
+            films = data['films']
+            # Grab username from data
             username = data['username']
-            user = CustomUser.objects.filter(username=username)
-            print(user)
-            new_data = data['films']
-            new_data['user'] = user[0]
-            film_results = FilmResults(**new_data)
+            # Find user by get
+            user = CustomUser.objects.get(username=username)
+            # Add user to films
+            films['user'] = user
+            # If movie doesn't have a rating add rating as -1 for distance function later
+            for film, ranking in films.items():
+                if ranking == '':
+                    films[film] = -1
+
+            matches = compute_user_matches(user)
+            for match in matches.keys():
+                user_to_match = get_object_or_404(CustomUser, id=match)
+
+                if user != user_to_match:
```

Comment:
This check is to avoid adding a match between a user and themself, yes? I thought this was avoided in the compute_user_matches.